hpy/devel/src/runtime/argparse.c

```diff
@@ -339,6 +346,36 @@ parse_item(HPyContext ctx, HPyTracker *ht, HPy current_arg, int current_arg_tmp,
 }
 
 
+/**
+ * Parse positional arguments.
+ *
+ * :param ctx:
+ *     The execution context.
+ * :param ht:
+ *     An optional pointer to an HPyTracker. If the format string never
+ *     results in new handles being created, `ht` may be `NULL`. Currently
+ *     no formatting options to this function require an HPyTracker.
+ * :param args:
+ *     The array of positional arguments to parse.
+ * :param nargs:
+ *     The number of elements in args.
+ * :param fmt:
+ *     The format string to use to parse the arguments.
+ * :param ...:
+ *     A va_list of references to variables in which to store the parsed
+ *     arguments. The number and types of the arguments should match the
+ *     the format strint, `fmt`.
+ *
+ * :returns: 0 on failure, 1 on success.
+ *
+ * If a `NULL` pointer is passed to `ht` and an `HPyTracker` is required by
+ * the format string, an exception will be raised.
+ *
+ * If a pointer is provided to `ht`, the `HPyTracker` will always be created
+ * and must be closed with `HPyTracker_Close` if parsing succeeds (after all
+ * handles returned are no longer needed). If parsing fails, this function
+ * will close the `HPyTracker` automatically.
```

Comment:
would be nice to have also an "Examples" section

Reply:
Also a good idea. I'll make an attempt at an Examples section in a bit.

Reply:
Examples added. Adding the example also made me realize we didn't have a test for passing an HPyTracker to HPyArg_Parse.